Description: Modifies the launcher scripts to specify the default Java and jtreg paths
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/src/share/bin/jtdiff.sh
+++ b/src/share/bin/jtdiff.sh
@@ -37,6 +37,10 @@
 #
 # jtdiff also provides an Ant task for direct invocation from Ant.
 
+if [ -z "${JT_HOME}" ]; then
+   JTREG_HOME=`readlink -f "${0}" | xargs dirname | xargs dirname`
+fi
+
 # Determine jtdiff/JavaTest installation directory
 if [ -n "$JT_HOME" ]; then
     if [ ! -r $JT_HOME/lib/jtreg.jar ];then
@@ -96,7 +100,7 @@
 elif [ -n "$JAVA_HOME" ]; then
     JT_JAVA="$JAVA_HOME/bin/java"
 else
-    JT_JAVA=java
+    JT_JAVA=/usr/lib/jvm/default-java/bin/java
 fi
 
 # Verify java version (1.)5 or newer used to run jtdiff
--- a/src/share/bin/jtreg.sh
+++ b/src/share/bin/jtreg.sh
@@ -70,6 +70,10 @@
 
 # Determine jtreg installation directory
 JTREG_HOME=${JTREG_HOME:-$JT_HOME}      # allow for old version of name
+if [ -z "${JTREG_HOME}" ]; then
+    JTREG_HOME=`readlink -f "${0}" | xargs dirname | xargs dirname`
+fi
+
 if [ -n "$JTREG_HOME" ]; then
     if [ ! -r $JTREG_HOME/lib/jtreg.jar ];then
         echo "Invalid JTREG_HOME=$JTREG_HOME. Cannot find or read $JTREG_HOME/lib/jtreg.jar"
@@ -125,7 +129,7 @@
 elif [ -n "$jdk" ]; then
     JTREG_JAVA="$jdk/bin/java"
 else
-    JTREG_JAVA=java
+    JTREG_JAVA=/usr/lib/jvm/default-java/bin/java
 fi
 
 # Fixup JTREG_JAVA, JTREG_HOME as needed, if using Cygwin or WSL
